server/src/main/java/org/elasticsearch/cluster/metadata/MetadataMappingService.java

```diff
@@ -179,19 +179,10 @@ private boolean refreshIndexMapping(IndexService indexService, IndexMetadata.Bui
         boolean dirty = false;
         String index = indexService.index().getName();
         try {
-            List<String> updatedTypes = new ArrayList<>();
             DocumentMapper mapper = indexService.mapperService().documentMapper();
-            if (mapper != null) {
-                final String type = mapper.type();
-                if (!mapper.mappingSource().equals(builder.mapping(type).source())) {
-                    updatedTypes.add(type);
-                }
-            }
-
-            // if a single type is not up-to-date, re-send everything
-            if (updatedTypes.isEmpty() == false) {
```

Comment:
Wasn't that preventing unnecessary updates?

Reply:
Yes but the logic is still the same. `updatedTypes` wasn't empty if `!mapper.mappingSource().equals(builder.mapping(type).source())` - the update block is still behind the same guard.